### ISSUE TITLE
issues with sorbet

### ISSUE THREAD

**Issue by modosc** [fxn/zeitwerk#118]
i noticed that [sorbet is 100% slower with :zeitwerk](https://github.com/sorbet/sorbet/issues/2834) than with `:classic` mode in a brand new rails 6 application. i have a bug opened up in the sorbet repo but i figured it couldn't hurt to reach out to you about this as well. 

do you see anything obvious in the stack traces (or in the [suspect code](https://github.com/sorbet/sorbet/blob/1f0647d69e565e6968a7a209139180b32c626794/gems/sorbet/lib/hidden-definition-finder.rb#L193)) that might help me figure out what's going on here?

**Comment by fxn** [comment]
Hey, which are the stack traces?

**Comment by fxn** [comment]
Ah! They are in the ticket linked first.

**Comment by fxn** [comment]
The issue in Sorbet does not seem to move by now.

I am not familiar with Sorbet (know what it is, but have not used it). Could you give me reproduction steps with a minimal app?

**Comment by modosc** [comment]
> I am not familiar with Sorbet (know what it is, but have not used it). Could you give me reproduction steps with a minimal app?

sure! all i did was build a brand new rails 6 app via `rails new`. then i added `sorbet` to the gemfile and ran `srb rbi init`. 

with `:zeitwork` mode enabled this takes twice as long as with `:classic` mode. and the `:classic` mode timing with rails 6 is more or less identical to the timing with a brand new rails 5.2 app. 

i narrowed it down to to one step, `srb rbi hidden-definitions` which i believe essentially loads _everything_ in the gemfile and rails app and then walks object space to find classes (and find methods on those classes) so it can generate a list of methods that haven't been typed yet.

[here's the entry point](https://github.com/sorbet/sorbet/blob/1f0647d69e565e6968a7a209139180b32c626794/gems/sorbet/lib/hidden-definition-finder.rb#L41) of the code, and you can see in the other issue how i narrowed it down to `serialize_constants`

i'm close to abandoning sorbet because of this issue - i can't ask my team to run a 2+ hour command every time they change code to keep sorbet's type checking happy :/

**Comment by fxn** [comment]
How strange, it does not square here.

First, I generated a new application without JavaScript, Spring, or Bootsnap. Running

```
/usr/bin/time bundle exec srb rbi init .
```

yielded consistently some 31s with `classic`, and around 25s for `zeitwerk`. So in my laptop it is the other way around, `zeitwerk` mode is faster.

Then I tried a new application with all defaults (which include Spring, and Bootsnap), and times went down to 25s for `classic` and 22s for `zeitwerk`. Still faster for `zeitwerk`.

So definitely not reproducing that 2x slow down.

Sorbet is 0.5.5575, is it the version you're using?
